### PR TITLE
adat output tests for smux > 1

### PR DESCRIPTION
**Pull request by shuchitak**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -203,7 +203,7 @@ pipeline {
                                 "usb_audio_xcai_exp_dut", "usb_audio_xcai_exp_harness"]) { xtagIds ->
                         sh "pytest -v --level ${params.TEST_LEVEL} --junitxml=pytest_result_mac_intel.xml \
                             -o xk_216_mc_dut=${xtagIds[0]} -o xk_216_mc_harness=${xtagIds[1]} \
-                            -o xk_evk_xu316_dut=${xtagIds[2]} -o xk_evk_xu316_harness=${xtagIds[3]}"
+                            -o xk_evk_xu316_dut=${xtagIds[2]} -o xk_evk_xu316_harness=${xtagIds[3]} -k 'test_adat or test_spdif' "
```
Comment (xross) [inline Jenkinsfile:206]: Are these temporary changes?

File: app_usb_aud_xk_316_mc/src/core/xua_conf.h [hunk 2]
```diff
@@ -98,7 +98,7 @@
     #if (I2S_CHANS_DAC + 2*XUA_SPDIF_TX_EN + 8*XUA_ADAT_TX_EN) <= NUM_USB_CHAN_OUT
         #define ADAT_TX_INDEX    (I2S_CHANS_DAC + 2*XUA_SPDIF_TX_EN)
     #else
-        #define ADAT_TX_INDEX    (0)
+        #define ADAT_TX_INDEX    (I2S_CHANS_DAC + 2*XUA_SPDIF_TX_EN)
```
Comment (xross) [inline app_usb_aud_xk_316_mc/src/core/xua_conf.h:101]: You might have to explain this one to me - ADAT_TX_INDEX can now be > NUM_USB_CHAN_OUT causing a trap?

**Comment by shuchitak** [comment]
> I started reviewing this as I got a notification, but I think maybe I shouldn't be! I think its work in progress. If you mark as draft then I'll ignore :)

Hey, yes, this is still WIP. I've marked it as draft now.

File: app_usb_aud_xk_316_mc/src/core/xua_conf.h (hunk 2 above)
Comment (shuchitak) [reply app_usb_aud_xk_316_mc/src/core/xua_conf.h]: from what I understand, ADAT_TX_INDEX should always be I2S_CHANS_DAC + 2*XUA_SPDIF_TX_EN.
The #if (I2S_CHANS_DAC + 2*XUA_SPDIF_TX_EN + 8*XUA_ADAT_TX_EN) <= NUM_USB_CHAN_OUT check assumes that there are 8 ADAT channels which is not true if we build configs such as min_freq=88, max_freq=96 and num_usb_chan_out=12. 
I can add a compile time check to ensure ADAT_TX_INDEX is less than NUM_USB_CHAN_OUT.

**Pull request merged**